### PR TITLE
UI minor tweaks

### PR DESCRIPTION
**Pull request by jean-emmanuel**
Here's an attempt to solve a few issues I encountered while working on the [arc theme](https://github.com/jean-emmanuel/ardour-arc-theme), and to make the ui a tiny bit more pixel perfect. I think these changes may help creating more consistent themes.

Updated changes summary:
- make some hard-coded black drawing customizable via theming (button/fader/meter outlines)
- make flat mode faders flatter
- make various alignment tweaks in mixer strips
- attempt to replace Gtk::Frame with EventBox in some places in order to draw solid borders instead of beveled shadows, probably the biggest change and the most problematic in regard to Mixbus.
- set consistent padding in mixer strips for a more relaxed layout
- fix some background color inconsistencies in the mixer window
- fix various interaction inconsistencies in the mixer (
- fix alignment of track separators / canvas separators in editor and make the former more visible

A few comparison screenshots:
- [before mixer](https://github.com/Ardour/ardour/assets/5261671/987583a0-1cd0-4292-a2ea-7d06b038f8b5) / [after mixer](https://github.com/Ardour/ardour/assets/5261671/e69cf73b-7cf1-40c2-9f63-0c148258ad0e)
- [before_detail](https://github.com/Ardour/ardour/assets/5261671/3550616d-72d5-429d-90bf-501bd22aa262) / [after_detail](https://github.com/Ardour/ardour/assets/5261671/bdd1e5b7-8bb0-4707-b58c-e3ce9d00de24)

### CODE REVIEW

**Comment by jean-emmanuel** [comment]
Well to me it feels a bit strange to see the rounded corner when the fader is at 0dB, but I could live with it :)

**Comment by x42** [comment]
yeah, I suppose with *flat* style it makes some sense.

**Comment by jean-emmanuel** [comment]
Added 1 commit to remove the tooltip's outline that used the same color as the generic selection color, it felt somewhat out of place.

**Comment by jean-emmanuel** [comment]
3 new commits added with detailed descriptions, I believe these are tiny steps toward a more seamless interface.

**Comment by x42** [comment]
keep in mind that things need to align with default settings:  disable flat, and boxy options
Check with Editor Region Name bars, those have different spacing in the editor (separator line at the bottom).
The mixer shadows can't go away or need `#ifdef MIXBUS` -- getting rid of the `global_frame` is likely not an option. At least not short/mid-term since it will require significant UI changes in derivative projects.
Increasing the spacing and borders is not ideal, at least not not 100% scaling with full HD. Are you using a 4k screen?
Number label spacing is also required to be 1px for things to line up, and keep in mind there is `Mixer_UI::screenshot` which repacks strips.

We can cherry-pick the outline color changes once the pending reviews are addressed.

The overall direction you work towards is fine, and many of the other changes would benefit from a discussion.

**Comment by jean-emmanuel** [comment]
Thanks for your feedback. I might have gone a little ahead of myself, but I guess I'm just curious to see how far I can smooth out what I see as _tiny_ layout imperfections. What I'm mostly trying to tackle are the visual artifacts of what seems to be the result of fighting with gtk2 to get things done (using Frame to create borders and then being stuck with the beveled shadow look and a fixed 2px border size is a good example I think), though I'm aware replacing a hack with another hack is not necessarily desirable. I also appreciate the overall consistency that's already present in ardour's UI and I don't wan't to be pushy with immature changes, the software is huge and I'm merely scratching the surface here.

> The mixer shadows can't go away or need #ifdef MIXBUS -- getting rid of the global_frame is likely not an option. At least not short/mid-term since it will require significant UI changes in derivative projects.

That's indeed something I didn't anticipate, do you think it makes some sense to keep working in this direction as a proof of concept to see if such changes could be worth it at some point  

> Increasing the spacing and borders is not ideal, at least not not 100% scaling with full HD. Are you using a 4k screen?

In the end I think the difference is very subtle if you compare both versions, but unfortunately I can't test it on a 4k screen. Is the problem that some things are scaled and some are not (like spacing / borders) ?

> Number label spacing is also required to be 1px for things to line up, and keep in mind there is Mixer_UI::screenshot which repacks strips.

Maybe I got it wrong or fixed it the wrong way, but it seemed to align better after my change. Completely forgot about `Mixer_UI::screenshot`...

> We can cherry-pick the outline color changes once the pending reviews are addressed.

If something seems relevant, please do. And if the rest never makes it upstream, I'd still have learned a few things on the way :).

File: libs/widgets/ardour_button.cc [hunk 1]
```diff
@@ -811,6 +812,9 @@ ArdourButton::set_colors ()
 		led_active_color = UIConfigurationBase::instance().color ("generic button: led active");
 	}
 
+	outline_color = UIConfigurationBase::instance().color ("generic button: outline");
+
```
[the remark this answers is not in the document]
Comment (jean-emmanuel) [reply libs/widgets/ardour_button.cc:816]: Thank you, this one is now fixed I think.

File: libs/widgets/fastmeter.cc [hunk 2]
```diff
@@ -126,6 +130,8 @@ FastMeter::FastMeter (long hold, unsigned long dimen, Orientation o, int len,
 	request_width = pixrect.width + 2;
 	request_height= pixrect.height + 2;
 
+	outline_color = UIConfigurationBase::instance().color ("generic button: outline");
```
[the remark this answers is not in the document]
Comment (jean-emmanuel) [reply libs/widgets/fastmeter.cc:133]: Done

File: libs/widgets/ardour_fader.cc [hunk 3]
```diff
@@ -81,6 +83,9 @@ ArdourFader::ArdourFader (Gtk::Adjustment& adj, int orientation, int fader_lengt
 	} else {
 		CairoWidget::set_size_request(_span, _girth);
 	}
+
+	outline_color = UIConfigurationBase::instance().color ("generic button: outline");
```
[the remark this answers is not in the document]
Comment (jean-emmanuel) [reply libs/widgets/ardour_fader.cc]: And done !

**Pull request closed without merging**